sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -298,7 +304,10 @@ private void decodeDictionaryIds(
             // TODO: Convert dictionary of Binaries to dictionary of Longs
             if (!column.isNullAt(i)) {
               Binary v = dictionary.decodeToBinary(dictionaryIds.getDictId(i));
-              column.putLong(i, ParquetRowConverter.binaryToSQLTimestamp(v));
+              long rawTime = ParquetRowConverter.binaryToSQLTimestamp(v);
+              long adjTime =
+                  convertTz == null ? rawTime : DateTimeUtils.convertTz(rawTime, convertTz, UTC);
```

Comment:
How about skipping conversion if `convertTz == UTC` as well?

Reply:
good idea.  I'll push that check further up, so UTC is replaced by null, so we do simpler checks inside here